### DISCUSSION TITLE
Datatype for historical Dates

### DISCUSSION THREAD

**Discussion by mafiore** [ClickHouse/ClickHouse#34896] [Ideas]
In our hospital we have to poke around with people's birthdates, which could not be represented with the different Datexxx types
(lots of the people stored our datawarehouse have birthdates before 1925 - a few of them are still alive)
We store these information as YYYY-MM-DD Strings  which is quite ok. It is a bit more tricky when it comes to arithmetic (correct age of a person  related to a timestamp  ) I wrote a UDF to solve this. 

It would be cool to have a Datatype "Historic" backed up with a bundle of related functions to work with such information.
It could have a second parameter describing the data's granularity. (day/year for birthdates, year for historic events, decade, century, millenium,.... ) to make it versatile for different use cases (extreme e.g. prehistoric research .. )

**Comment by den-crane** [comment]
related https://github.com/ClickHouse/ClickHouse/issues/32006

I guess the accuracy of the age is not important? +/-13 days ?

**Reply by mafiore** [reply]
For older birthdates the precicion of years is exact enough. for the newer ones it have to be day correct

**Reply by mafiore** [reply]
with older birthdates I mean before 1970 :-). The Proleptic Gregorian calendar seems to be a cool idea
